.circleci/config.yml

```diff
@@ -30,60 +34,99 @@ common_filters: &common_filters
       ignore:
         - gh-pages
 
+common_jobs: &common_jobs
+    jobs:
+        - find-develop-snapshot
+        - build:
+            <<: *common_filters
+            requires:
+              - find-develop-snapshot
+        - unit-tests:
+            <<: *common_filters
+            requires:
+              - build
+        - integration-tests:
+            matrix:
+              parameters:
+                testing_profile: [ "singularity-tests", "bitbucket-tests" ]
+            <<: *common_filters
+            requires:
+              - build
+        - non-confidential-tests:
+            <<: *common_filters
+            requires:
+              - build
+        - confidential-workflow-tests:
+            <<: *common_filters
+            requires:
+              - build
+        - confidential-tool-tests:
+            <<: *common_filters
+            requires:
+              - build
+        - wes-toil-test:
+            <<: *common_filters
+            requires:
+              - build
+        - sonar-cloud:
+            <<: *common_filters
+            requires:
+              - unit-tests
+              - integration-tests
+              - non-confidential-tests
+              - confidential-workflow-tests
+              - confidential-tool-tests
+              - wes-toil-test
+            context:
+              - sonarcloud
+
 workflows:
   version: 2
+
   everything:
-    jobs:
-      - build:
-          <<: *common_filters
-
-      - unit-tests:
-          <<: *common_filters
-          requires:
-            - build
-      - integration-tests:
-          matrix:
-            parameters:
-              testing_profile: ["singularity-tests", "bitbucket-tests"]
-          <<: *common_filters
-          requires:
-            - build
-      - non-confidential-tests:
-          <<: *common_filters
-          requires:
-            - build
-      - confidential-workflow-tests:
-          <<: *common_filters
-          requires:
-            - build
-      - confidential-tool-tests:
-          <<: *common_filters
-          requires:
-            - build
-      - wes-toil-test:
-          <<: *common_filters
-          requires:
-            - build
-      - sonar-cloud:
-          <<: *common_filters
-          requires:
-            - unit-tests
-            - integration-tests
-            - non-confidential-tests
-            - confidential-workflow-tests
-            - confidential-tool-tests
-            - wes-toil-test
-          context:
-            - sonarcloud
+    when:
+      not: << pipeline.parameters.run_against_develop_core >>
+    <<: *common_jobs
 
+  nightly:
```

Comment:
The reason I created two workflows, instead of having just one controlled by the flag (my first attempt), is because if the `nightly` build runs while the `everything` build is running, the `everything` build will get cancelled.